packages/components/pie-modal/src/index.ts

```diff
@@ -1,18 +1,38 @@
-import { LitElement, html, unsafeCSS } from 'lit'; // eslint-disable-line import/no-extraneous-dependencies
-import { property } from 'lit/decorators.js';
-import { RtlMixin } from '@justeattakeaway/pie-webc-core';
+import { LitElement, unsafeCSS } from 'lit'; // eslint-disable-line import/no-extraneous-dependencies
+import { html, unsafeStatic } from 'lit/static-html.js';
+import { property } from 'lit/decorators.js'; // eslint-disable-line import/no-extraneous-dependencies
+import { RtlMixin, validPropertyValues, requiredProperty } from '@justeattakeaway/pie-webc-core';
 
 import styles from './modal.scss?inline';
+import { HEADING_LEVELS } from './defs';
+
+// Valid values available to consumers
+export { HEADING_LEVELS };
+
+const componentSelector = 'pie-modal';
 
 export class PieModal extends RtlMixin(LitElement) {
     @property({ type: Boolean })
         isOpen = false;
 
-    // eslint-disable-next-line class-methods-use-this
+    @property({ type: String })
+    @requiredProperty(componentSelector)
+        heading: string | undefined;
```

Comment:
if it's required then I don't think it can be `undefined`, right?

Reply:
I agree, but I get this error if I don't specify if the component potentially won't be assigned, do you have any idea?
<img width="708" alt="Screenshot 2023-06-09 at 16 43 03" src="https://github.com/justeattakeaway/pie/assets/28605803/3b608359-38ae-4bb9-ab7f-cc23ea1a48f3">

Reply:
I don't have an immediate suggestion but maybe we can look into this a bit more as I think it could make the codebase messy if we set a precedent here

Reply:
I did a bit of digging and I believe we can use something called _definite assignment assertion operator_ or `!` in short to achieve a semi-required behavior in classes and ts. 

assuming the title is a required field and should always be string, setting `heading!: string` will tell the compiler that the heading property will be instantiated with a value before it's used, even though it might not be apparent for you now, but every time we get or set that property, we want it to be treated it as a type of string. 👓